src/Aspire.Dashboard/Components/Pages/IMemoryPage.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Server.ProtectedBrowserStorage;
+
+namespace Aspire.Dashboard.Components.Pages;
+
+public interface IMemoryPage<in T>
```

Comment:
I'm not sure about the name of this interface, and am struggling to come up with something else. I think it's because the interface isn't so self describing to me.

Some of my confusion is perhaps not knowing what state we realistically want to store in local storage vs. the URL itself. It seems like putting state in the URL is the better approach, as it works naturally with browser actions (sharing, bookmarks, etc).

Perhaps if you write some API docs for this type you'll come up with a better name in the process, or those docs would at least help me understand the "personality" of this interface.